l1-contracts/contracts/bridge/L1NativeTokenVault.sol

```diff
@@ -0,0 +1,161 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.24;
+
+import {Ownable2StepUpgradeable} from "@openzeppelin/contracts-upgradeable/access/Ownable2StepUpgradeable.sol";
+import {PausableUpgradeable} from "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
+
+import {IERC20Metadata} from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import {IL1NativeTokenVault} from "./interfaces/IL1NativeTokenVault.sol";
+import {ReentrancyGuard} from "../common/ReentrancyGuard.sol";
+import {IL1StandardAsset} from "./interfaces/IL1StandardAsset.sol";
+
+import {IL1SharedBridge} from "./interfaces/IL1SharedBridge.sol";
+import {ETH_TOKEN_ADDRESS, TWO_BRIDGES_MAGIC_VALUE} from "../common/Config.sol";
+
+import {IBridgehub, L2TransactionRequestTwoBridgesInner, L2TransactionRequestDirect} from "../bridgehub/IBridgehub.sol";
+
+/// @author Matter Labs
+/// @custom:security-contact security@matterlabs.dev
+/// @dev Vault holding L1 native ETH and ERC20 tokens bridged into the hyperchains.
+/// @dev Designed for use with a proxy for upgradability.
+contract L1NativeTokenVault is
```

Comment:
I can see that this essentially separates from the shared bridge, but in the diagram we have a separation between native token vault & the standard scenario.

Does this mean we're only considering here the tokens native to L1?
As I understand the vault for L2 native tokens has yet to be added

Reply:
Yes, we cannot handle l2 native token yet. ( You could implement it if you want )